src/main/java/seedu/partyplanet/logic/commands/EditToRemoveTagCommand.java

```diff
@@ -34,45 +33,57 @@ public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
         List<Person> lastShownList = model.getFilteredPersonList();
 
-        for (Person person: lastShownList) {
-            removeTagFromPerson(model, person);
+        for (Person person : lastShownList) {
+            checkIfHasTags(person);
         }
 
         if (editedPersons.isEmpty()) {
             return new CommandResult(MESSAGE_TAGS_NOT_REMOVED);
         }
 
+        for (Person personToEdit : editedPersons) {
+            removeTagFromPerson(model, personToEdit);
+        }
+
         model.addState();
-        model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
 
         return new CommandResult(String.format(MESSAGE_REMOVE_TAGS_SUCCESS,
                 editedPersons.isEmpty() ? "" : displayPersons(editedPersons)));
     }
 
-    private void removeTagFromPerson(Model model, Person personToEdit) {
-        Set<Tag> personTags = new HashSet<>(personToEdit.getTags());
+    private void checkIfHasTags(Person personToCheck) {
+        Set<Tag> personTags = personToCheck.getTags();
 
         boolean isEdited = false;
-        for (Tag targetTag: targetTags) {
+        for (Tag targetTag : targetTags) {
             if (isEdited) { // prevent true from reverting back to false
-                personTags.remove(targetTag);
             } else {
-                isEdited = personTags.remove(targetTag);
+                isEdited = personTags.contains(targetTag);
             }
         }
         if (isEdited) {
-            editedPersons.add(personToEdit);
-            Person editedPerson = new Person(personToEdit.getName(), personToEdit.getPhone(), personToEdit.getEmail(),
-                    personToEdit.getBirthday(), personToEdit.getAddress(), personToEdit.getRemark(), personTags);
-            model.setPerson(personToEdit, editedPerson);
+            editedPersons.add(personToCheck);
         }
     }
```

Comment:
Alternative to probably make it simpler and slightly more efficient

```
    private void checkIfHasTags(Person personToCheck) {
        Set<Tag> personTags = personToCheck.getTags();

        for (Tag targetTag : targetTags) {
            if (personTags.contains(targetTag)) {
                editedPersons.add(personToCheck);
                return;
            }
        }
    }
```

Reply:
Adopted in latest commit